File: python/program/Laptop.py
# Import class Komputer dari file Komputer.py
from Komputer import Komputer


class Laptop(Komputer):
    """
    Class untuk merepresentasikan Laptop, merupakan turunan dari Komputer.
    """

    def __init__(
        self,
        # Atribut dari Elektronik
        id_produk: str = "",
        nama_produk: str = "",
        harga_produk: int = 0,
        stok_produk: int = 0,
        foto_produk: str = "",
        # Atribut dari Komputer
        motherboard: str = "",
        cpu: str = "",
        gpu: str = "",
        ram: str = "",
        storage: str = "",
        # Atribut milik Laptop sendiri
        baterai: str = "",
        ukuran_layar: str = "",
        resolusi_layar: str = "",
    ):
        # Panggil constructor dari parent class (Komputer) untuk inisialisasi semua atribut di atasnya
        super().__init__(
            id_produk,
            nama_produk,
            harga_produk,
            stok_produk,
            foto_produk,
            motherboard,
            cpu,
            gpu,
            ram,
            storage,
        )

        # Inisialisasi atribut milik Laptop
        self.baterai = baterai
        self.ukuran_layar = ukuran_layar
        self.resolusi_layar = resolusi_layar

    # --- Getters ---
    def get_baterai(self) -> str:
        return self.baterai

    def get_ukuran_layar(self) -> str:
        return self.ukuran_layar

    def get_resolusi_layar(self) -> str:
        return self.resolusi_layar

    # --- Setters ---
    def set_baterai(self, value: str):
        self.baterai = value

    def set_ukuran_layar(self, value: str):
        self.ukuran_layar = value

    def set_resolusi_layar(self, value: str):
        self.resolusi_layar = value

File: php/program/Laptop.php
<?php

require_once 'Komputer.php';

class Laptop extends Komputer {
    private string $baterai;
    private string $ukuran_layar;
    private string $resolusi_layar;

    public function __construct(
        string $id_produk = "", string $nama_produk = "", int $harga_produk = 0, int $stok_produk = 0, string $foto_produk = "",
        string $motherboard = "", string $cpu = "", string $gpu = "", string $ram = "", string $storage = "",
        string $baterai = "", string $ukuran_layar = "", string $resolusi_layar = ""
    ) {
        parent::__construct(
            $id_produk, $nama_produk, $harga_produk, $stok_produk, $foto_produk,
            $motherboard, $cpu, $gpu, $ram, $storage
        );
        $this->baterai = $baterai;
        $this->ukuran_layar = $ukuran_layar;
        $this->resolusi_layar = $resolusi_layar;
    }

    // --- Getters (Pastikan namanya seperti ini) ---
    public function getBaterai(): string {
        return $this->baterai;
    }

    public function getUkuran_layar(): string { // diganti menjadi snake_case
        return $this->ukuran_layar;
    }

    public function getResolusi_layar(): string { // diganti menjadi snake_case
        return $this->resolusi_layar;
    }

    // --- Setters ---
    public function setBaterai(string $baterai): void {
        $this->baterai = $baterai;
    }

    public function setUkuran_layar(string $ukuran_layar): void { // diganti menjadi snake_case
        $this->ukuran_layar = $ukuran_layar;
    }

    public function setResolusi_layar(string $resolusi_layar): void { // diganti menjadi snake_case
        $this->resolusi_layar = $resolusi_layar;
    }
}
